src/krb5_plugin/passkey/passkey_clpreauth.c

```diff
@@ -365,6 +365,9 @@ sss_passkeycl_process(krb5_context context,
     if (reply != NULL) {
         free(reply);
     }
+    if (prompt_reply != NULL) {
+        free(prompt_reply);
+    }
```

Comment:
Since C99 standard if the pointer passed to free is NULL, then no action occurs ([ISO/IEC 9899:1999](https://www.dii.uchile.cl/~daespino/files/Iso_C_1999_definition.pdf)).

@alexey-tikhonov can we assume that we are using this or a higher standard for sssd? [shadow-utils](https://github.com/shadow-maint/shadow/issues/600) already assumed it last year, and Linux-PAM is working on it. This would help us remove some redundant code, and maybe even start using functions that are provided in higher standards.

Reply:
We actually require C11 due to usage of atomic types.

Reply:
In that case let's remove the condition

Reply:
Done.